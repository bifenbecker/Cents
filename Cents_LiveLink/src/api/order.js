import httpClient from "./httpClient";

export const fetchStoreSettings = params => {
  return httpClient({
    method: "GET",
    url: "/live-status/settings",
    params,
  });
};

export const authenticateOrder = token => {
  return httpClient({
    method: "GET",
    url: "/live-status/verify-order",
    params: {
      token,
    },
  });
};

export const fetchOrderDetail = token => {
  return httpClient({
    method: "GET",
    url: "/live-status",
    params: {
      token,
    },
  });
};

export const updateTip = (token, tipData) => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/tip",
    params: {
      token,
    },
    data: tipData,
  });
};

export const addPromo = (token, promoData) => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/add-promotion",
    params: {
      token,
    },
    data: promoData,
  });
};

export const removePromo = token => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/remove-promotion",
    params: {
      token,
    },
  });
};

export const addCredits = (token, creditsData) => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/add-credits",
    params: {
      token,
    },
    data: creditsData,
  });
};

export const removeCredits = token => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/remove-credits",
    params: {
      token,
    },
  });
};

export const setOnlineOrderReturnMethod = (token, data) => {
  return httpClient({
    method: "PUT",
    url: "/live-status/return-method",
    params: {
      token,
    },
    data,
  });
};

export const updatePaymentMethodIntent = (token, {paymentToken}) => {
  return httpClient({
    method: "PUT",
    url: "/live-status/payment-method",
    params: {
      token,
    },
    data: {paymentToken},
  });
};

export const cancelOrder = params => {
  return httpClient({
    method: "PATCH",
    url: "/live-status/order/cancel",
    params,
  });
};

export const getRouteDelivery = (token, {orderDelivery}) => {
  return httpClient({
    method: "GET",
    url: `/live-status/order-delivery/${orderDelivery}/route-details`,
    params: {token},
  });
};

export const fetchOrderDeliveries = token => {
  return httpClient({
    method: "GET",
    url: `/live-status/order-deliveries`,
    params: {token},
  });
};

export const manageOrder = (token, data) => {
  return httpClient({
    method: "PUT",
    url: `/live-status/live-link/manage`,
    params: {token},
    data,
  });
};
