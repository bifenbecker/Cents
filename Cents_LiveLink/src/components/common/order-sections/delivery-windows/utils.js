import {ORDER_DELIVERY_TYPES} from "../../../../constants/order";
import {getLuxonWeekDayFromMillis, isSameTimeForWindows} from "../../../../utils/date";

const isSameAsWindowAsSubscription = (subscription, orderDelivery, timeZone) => {
  const isPickup = orderDelivery.type === ORDER_DELIVERY_TYPES.pickup;
  const {deliveryTimingsId, returnWindow, pickupTimingsId, pickupWindow} = subscription;
  const susbcriptionTimingsId = isPickup ? pickupTimingsId : deliveryTimingsId;
  const subscriptionWindow = isPickup ? pickupWindow : returnWindow;

  const subscriptionWeekDay = getLuxonWeekDayFromMillis(
    subscriptionWindow?.[0],
    timeZone
  );
  const weekDay = getLuxonWeekDayFromMillis(orderDelivery?.deliveryWindow?.[0], timeZone);

  return (
    (orderDelivery?.timingsId || null) === susbcriptionTimingsId &&
    weekDay === subscriptionWeekDay &&
    isSameTimeForWindows(
      orderDelivery?.deliveryWindow || [],
      subscriptionWindow || [],
      timeZone
    )
  );
};

export const checkIsSubscriptionWithSameTime = (
  newOrderDelivery,
  subscriptionList,
  timeZone
) => {
  return subscriptionList.some(subscription => {
    return (
      isSameAsWindowAsSubscription(subscription, newOrderDelivery?.pickup, timeZone) &&
      isSameAsWindowAsSubscription(subscription, newOrderDelivery?.delivery, timeZone)
    );
  });
};
