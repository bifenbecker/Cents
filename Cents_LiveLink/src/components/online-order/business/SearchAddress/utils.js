import {useCallback} from "react";
import {geocodeByAddress, getLatLng} from "react-places-autocomplete";
import {getTimezoneFromLatLng} from "../../../../api/google";

export const mapGoogleAddressResult = (result) => {
  const address = {
    googlePlacesId: result.place_id,
    address1: null,
    city: null,
    firstLevelSubdivisionCode: null,
    postalCode: null,
    countryCode: null,
  };

  const address1Fields = [
    result.address_components.find((comp) => comp.types.includes("street_number"))
      ?.long_name,
    result.address_components.find((comp) => comp.types.includes("route"))?.long_name,
  ].filter((v) => v);

  // Both street_number and route are mandatory
  address.address1 = address1Fields.length === 2 ? address1Fields.join(" ") : null;

  address.city = result.address_components.find(
    (comp) =>
      comp.types.includes("locality") ||
      comp.types.includes("sublocality") ||
      comp.types.includes("administrative_area_level_3") ||
      comp.types.includes("administrative_area_level_2")
  )?.long_name;

  address.firstLevelSubdivisionCode = result.address_components.find((comp) =>
    comp.types.includes("administrative_area_level_1")
  )?.short_name;

  address.postalCode = result.address_components.find((comp) =>
    comp.types.includes("postal_code")
  )?.short_name;

  address.countryCode = result.address_components.find((comp) =>
    comp.types.includes("country")
  )?.short_name;

  return address;
};

export const getAddressAndLatLng = async (address) => {
  try {
    const [addressResult] = await geocodeByAddress(address);
    const latLng = await getLatLng(addressResult);
    const timeZoneRes = await getTimezoneFromLatLng(latLng);

    return {addressResult, latLng, timeZone: timeZoneRes?.data?.timeZoneId};
  } catch (error) {
    console.error("Error", error);
    throw new Error(
      error === "ZERO_RESULTS"
        ? "Address not found"
        : "Could not get relevant address details"
    );
  }
};

export const getFormattedAddress = (address) => {
  if (Object.keys(address).length !== 0) {
    const {address1, city, firstLevelSubdivisionCode, postalCode} = address;
    return `${
      address1 ? `${address1}, ` : ""
    }${city}, ${firstLevelSubdivisionCode} ${postalCode}`;
  } else {
    return "";
  }
};
