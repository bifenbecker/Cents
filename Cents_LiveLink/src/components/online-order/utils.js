import {DateTime} from "luxon";

import {bufferRequiredForOrder} from "./constants";
import {DELIVERY_PROVIDERS} from "../../constants/order";
import {FETCHING_STATUS} from "constants/api";

export {formatAddress, createAddressPayload} from "../../utils/address";

export const toDollars = (amount) => {
  return `$${amount?.toFixed(2) || "0.00"}`;
};

export const getServicePriceBreakdown = (service) => {
  const {prices, hasMinPrice} = service;
  const priceString = getPriceString(service);

  let minPriceString;
  let isShortMinPriceDescription = false;

  if (hasMinPrice) {
    const {storePrice, minQty, minPrice} = prices[0];

    isShortMinPriceDescription =
      (Number(minQty || 0) * Number(storePrice || 0)).toFixed(2) ===
      Number(minPrice || 0).toFixed(2);
    minPriceString = isShortMinPriceDescription
      ? isShortMinPriceDescription && minQty
        ? `(min. ${minQty}lbs)`
        : ""
      : `Min: First ${minQty}lbs @ ${toDollars(minPrice)}`;
  }

  return {priceString, minPriceString, isShortMinPriceDescription};
};

export const getPriceString = ({serviceCategory, prices}) => {
  const {storePrice} = prices[0];
  return `${toDollars(storePrice)} / ${
    serviceCategory.category === "PER_POUND" ? "lb" : "unit"
  }`;
};

export const getMinPriceString = ({prices}) => {
  const {minPrice} = prices[0];
  return toDollars(minPrice);
};

export const getTimeFromMilliSeconds = (timeInMilliSeconds, timeZone) => {
  return DateTime.fromMillis(Number(timeInMilliSeconds)).setZone(timeZone);
};

export const isValidReturnWindowForProvider = (
  pickupStartWindow,
  deliveryWindow,
  turnAroundInHours,
  timeZone
) => {
  if (!deliveryWindow?.length) {
    return false;
  }
  const pickupTime = getTimeFromMilliSeconds(Number(pickupStartWindow), timeZone);
  const deliveryScheduleTime = getTimeFromMilliSeconds(deliveryWindow[0], timeZone);
  const minDiff = deliveryScheduleTime?.diff(pickupTime, "minutes")?.toObject()?.minutes;
  return minDiff >= turnAroundInHours * 60;
};

export const isValidWindowForProvider = (deliveryWindow, deliveryProvider, timeZone) => {
  const shiftTime = getTimeFromMilliSeconds(
    deliveryWindow[deliveryProvider === DELIVERY_PROVIDERS.ownDriver ? 0 : 1],
    timeZone
  );
  const currentTime = DateTime.local().setZone(timeZone);
  const minDiff = shiftTime?.diff(currentTime, "minutes")?.toObject()?.minutes || 0;
  return minDiff > bufferRequiredForOrder[deliveryProvider];
};

export const getAvailableOwnDriverOrOnDemandStoreId = (
  ownDeliveryStore,
  onDemandDeliveryStore
) => {
  if (ownDeliveryStore?.storeId) {
    return ownDeliveryStore?.storeId;
  }

  if (isOnDemandEnabled(onDemandDeliveryStore)) {
    return onDemandDeliveryStore?.storeId;
  }
};

export const isOnDemandEnabled = (onDemandDeliveryStore) => {
  const doordashStores = process.env.REACT_APP_DOORDASH_STORES.split(",");

  return (
    onDemandDeliveryStore?.storeId &&
    onDemandDeliveryStore?.active &&
    onDemandDeliveryStore?.dayWiseWindows?.length &&
    (onDemandDeliveryStore?.doorDashEnabled ||
      doordashStores?.includes(onDemandDeliveryStore?.storeId?.toString()))
  );
};

export const getTitleState = (fetchingStatus, isFirstVisit) => {
  if (fetchingStatus === FETCHING_STATUS.FULFILLED) {
    if (isFirstVisit) {
      return "firstVisit";
    } else {
      return "greeting";
    }
  } else {
    return "firstVisit";
  }
};
