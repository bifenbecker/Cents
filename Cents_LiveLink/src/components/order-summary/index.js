import React, {useEffect, useState} from "react";
import {useParams} from "react-router-dom";
import {Box, Flex, Text} from "rebass/styled-components";

// Components
import {Header} from "./../common";
import DeliveryOverview from "../delivery/DeliveryOverview";

import {fetchOrderDetail} from "../../api";
import {getPriceString} from "../../utils";

const OrderSummary = () => {
  const {orderToken} = useParams();
  const [orderDetails, setOrderDetails] = useState();
  const [loading, setLoading] = useState(true);
  const [loadingErrorMsg, setLoadingErrorMsg] = useState();
  const [showOrderDetails, setShowOrderDetails] = useState();
  const [showDeliveryModal, setShowDeliveryModal] = useState(false);

  useEffect(() => {
    fetchOrderDetail(orderToken)
      .then(({data: {orderDetails}}) => {
        setOrderDetails(orderDetails);
        setLoading(false);
        setShowOrderDetails(true);
      })
      .catch(error => {
        const errMsg = error.response?.data.error || error.message;
        setLoadingErrorMsg(errMsg);
      });
  }, [orderToken]);

  const renderHeader = () => {
    return (
      <Flex {...styles.header.wrapper}>
        <Box>
          <Text {...styles.header.title}>Order Summary</Text>
        </Box>
        {orderDetails.status === "COMPLETED" ? (
          <Text {...styles.header.status}>COMPLETE</Text>
        ) : null}
      </Flex>
    );
  };

  const renderCustomerDetail = () => {
    const {
      customer: {fullName, phoneNumber, email},
      store: {name, address, city, zipCode},
    } = orderDetails;
    return (
      <Box {...styles.address.wrapper}>
        <Text {...styles.address.info}>
          {fullName}
          <br />
          {phoneNumber}
          <br />
          {email}
          <br />
          {name} <br />
          {address}, {city} {zipCode}
        </Text>
      </Box>
    );
  };

  const renderOrderItems = () => {
    return (
      <Box {...styles.orderItems.wrapper}>
        {orderDetails.orderItems?.map(item => (
          <Box {...styles.orderItems.itemWrapper} key={item.orderItemId}>
            <Flex {...styles.orderItems.itemTitle}>
              <Text>{item.laundryType}</Text>
              <Text>${item.itemTotal?.toFixed(2) || "0.00"}</Text>
            </Flex>
            <Flex {...styles.orderItems.itemPriceInfo}>
              {getPriceString(item)
                .split("|")
                .map(text => (
                  <Text key={text}>{text}</Text>
                ))}
            </Flex>
          </Box>
        ))}
      </Box>
    );
  };

  const renderFooter = () => {
    return (
      <Box {...styles.footer.wrapper}>
        <Flex {...styles.footer.paidInfo}>
          <Text>Total Paid</Text>
          <Text>${orderDetails.totalPaid.toFixed(2)}</Text>
        </Flex>
        {orderDetails.tipAmount ? (
          <Flex {...styles.footer.otherPaidInfo}>
            <Text>Tip</Text>
            <Text>${orderDetails.tipAmount.toFixed(2)}</Text>
          </Flex>
        ) : null}
        {orderDetails.isTaxable ? (
          <Flex {...styles.footer.otherPaidInfo}>
            <Text>Tax</Text>
            <Text>${orderDetails.taxAmount.toFixed(2)}</Text>
          </Flex>
        ) : null}
        {orderDetails.creditAmount ? (
          <Flex {...styles.footer.otherPaidInfo}>
            <Text>Credit applied</Text>
            <Text>${orderDetails.creditAmount.toFixed(2)}</Text>
          </Flex>
        ) : null}
        <Flex {...styles.footer.dueInfo}>
          <Text>Total Due</Text>
          <Text>${orderDetails.balanceDue?.toFixed(2) || "0.00"}</Text>
        </Flex>
      </Box>
    );
  };

  const displayDeliveryModal = async () => {
    setShowDeliveryModal(true);
    setShowOrderDetails(false);
  };

  return (
    <Box>
      {/* <Header onDeliveryClick={displayDeliveryModal} /> */}
      {loading && (
        <Flex {...styles.loader.wrapper}>
          <Header onDeliveryClick={displayDeliveryModal} />
          <Text>{loadingErrorMsg ? loadingErrorMsg : "Loading ..."}</Text>
        </Flex>
      )}
      {!loading && (
        <Box>
          {showOrderDetails && (
            <Box>
              <Header order={orderDetails} onDeliveryClick={displayDeliveryModal} />
              {renderHeader()}
              {renderCustomerDetail()}
              {renderOrderItems()}
              {renderFooter()}
            </Box>
          )}
          {showDeliveryModal && (
            <DeliveryOverview
              order={orderDetails}
              closeModal={() => {
                setShowDeliveryModal(false);
                setShowOrderDetails(true);
              }}
            />
          )}
        </Box>
      )}
    </Box>
  );
};

const styles = {
  loader: {
    wrapper: {
      p: 3,
      height: "80vh",
      textAlign: "center",
      justifyContent: "center",
      flexDirection: "column",
    },
  },
  header: {
    wrapper: {
      p: 3,
      justifyContent: "space-between",
    },
    title: {
      fontSize: [3, 4],
      fontWeight: "500",
    },
    status: {
      fontSize: [0, 1],
      fontWeight: "600",
      bg: "HUB_NOTIFICATION_GREY",
      px: 2,
      py: 1,
      color: "CENTS_BLUE",
      sx: {
        borderRadius: 9999,
      },
    },
  },
  address: {
    wrapper: {
      sx: {
        borderBottom: "1px solid",
        borderColor: "DISABLED_TEXT_GREY",
      },
    },
    info: {
      pb: 4,
      px: 3,
      as: "address",
      fontSize: [0, 1],
      fontStyle: "normal",
    },
  },
  orderItems: {
    wrapper: {
      p: 3,
      minHeight: "60vh",
    },
    itemWrapper: {
      pb: 3,
    },
    itemTitle: {
      py: 1,
      fontSize: [1, 2],
      justifyContent: "space-between",
      fontWeight: "bold",
    },
    itemPriceInfo: {
      py: 1,
      fontSize: [1, 2],
      lineHeight: 1.5,
      flexDirection: "column",
    },
  },
  footer: {
    wrapper: {
      as: "footer",
      p: 3,
      py: [3, 4],
      width: 1,
      bg: "WHITE",
      fontWeight: "bold",
      sx: {
        position: "sticky",
        bottom: 0,
        boxShadow: "0 -5px 8px -7px rgba(0,0,0,.2)",
      },
    },
    paidInfo: {
      pb: [1, 2],
      fontSize: [1, 2],
      justifyContent: "space-between",
      lineHeight: 2,
    },
    dueInfo: {
      fontSize: [2, 3],
      justifyContent: "space-between",
      lineHeight: 1.6,
    },
    otherPaidInfo: {
      pb: [1, 2],
      fontSize: [1, 2],
      justifyContent: "space-between",
      lineHeight: 2,
      fontWeight: "normal",
    },
  },
};

export default OrderSummary;
