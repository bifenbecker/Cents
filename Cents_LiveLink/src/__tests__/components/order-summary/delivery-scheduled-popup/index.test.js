import React from "react";
import {fireEvent, render, waitFor} from "@testing-library/react";
import {act} from "react-dom/test-utils";

import DeliveryScheduledPopup from "../../../../components/order-summary/delivery-scheduled-popup";

import {
  ownDriverDelivery,
  onDemandDelivery,
} from "../../../../mocks/factories/orderDeliveries";
import {orderToken, orderId, timeZone} from "../../../../mocks/factories/order";
import {shiftDetails} from "../../../../components/order-summary/utils";
import {
  getDeliveryAcknowledgementKey,
  getParsedLocalStorageData,
} from "../../../../utils/common";

const onShowDeliveryOptionsClick = jest.fn();
const toggle = jest.fn();

const renderDeliveryScheduledPopup = props => {
  const {getByText, container, getByAltText, getByRole} = render(
    <DeliveryScheduledPopup
      {...{
        isOpen: true,
        toggle,
        onShowDeliveryOptionsClick,
        orderToken,
        orderId,
        orderDelivery: props.orderDelivery,
        timeZone,
      }}
    />
  );

  return {getByText, getByAltText, getByRole, container};
};

describe("DeliveryScheduledPopup Component", () => {
  describe("When deliveryProvider is OWN DRIVER", () => {
    it("should match snaphot", () => {
      const {container} = renderDeliveryScheduledPopup({
        orderDelivery: ownDriverDelivery,
      });

      expect(container).toMatchSnapshot();
    });

    it("should render van icon when the delivery provider is own driver", () => {
      const {getByAltText} = renderDeliveryScheduledPopup({
        orderDelivery: ownDriverDelivery,
      });

      const imageAltText = "Van Image";
      const icon = getByAltText(imageAltText);

      expect(icon).toBeInTheDocument();
    });
  });

  describe("When deliveryProvider is ON DEMAND", () => {
    it("should match snaphot", () => {
      const {container} = renderDeliveryScheduledPopup({
        orderDelivery: onDemandDelivery,
      });

      expect(container).toMatchSnapshot();
    });

    it("should render car icon when the delivery provider is On demand", () => {
      const {getByAltText} = renderDeliveryScheduledPopup({
        orderDelivery: onDemandDelivery,
      });

      const imageAltText = "Car Image";
      const icon = getByAltText(imageAltText);

      expect(icon).toBeInTheDocument();
    });
  });

  it("should render header text", () => {
    const {getByText} = renderDeliveryScheduledPopup({
      orderDelivery: ownDriverDelivery,
    });

    expect(getByText("Your laundry is ready!")).toBeInTheDocument();
  });

  it("should render see my delivery options text", () => {
    const {getByText} = renderDeliveryScheduledPopup({
      orderDelivery: ownDriverDelivery,
    });

    expect(getByText("See my delivery options")).toBeInTheDocument();
  });

  it("should render selected delivery window details", () => {
    const {getByText} = renderDeliveryScheduledPopup({
      orderDelivery: ownDriverDelivery,
    });
    const shiftInfo = shiftDetails(ownDriverDelivery, timeZone);

    expect(getByText("Delivery scheduled for")).toBeInTheDocument();
    expect(
      getByText(`${shiftInfo?.startTime} - ${shiftInfo?.endTime}`)
    ).toBeInTheDocument();
    expect(
      getByText(`${shiftInfo?.day}, ${shiftInfo?.month} ${shiftInfo?.date}`)
    ).toBeInTheDocument();
  });

  it("should trigger toggle with  when got it button is clicked", () => {
    const {getByText} = renderDeliveryScheduledPopup({
      orderDelivery: ownDriverDelivery,
    });

    const button = getByText("got it");
    fireEvent.click(button);

    expect(toggle).toBeCalled();

    const parsedAcknowledgementData = getParsedLocalStorageData(
      getDeliveryAcknowledgementKey(orderId)
    );
    expect(parsedAcknowledgementData).toHaveProperty("value");
    expect(parsedAcknowledgementData.value).toEqual(orderToken);
    expect(parsedAcknowledgementData).toHaveProperty("expiry");
    expect(parsedAcknowledgementData.expiry).toBeGreaterThan(new Date().getTime());
  });

  it("should call onShowDeliveryOptionsClick with currentOrder as selected choice", () => {
    const {getByText} = renderDeliveryScheduledPopup({
      orderDelivery: ownDriverDelivery,
    });
    fireEvent.click(getByText("See my delivery options"));

    expect(onShowDeliveryOptionsClick).toHaveBeenCalled();
  });

  it("should set fontSize of scheduled details text according to window width", async () => {
    const {getByText} = renderDeliveryScheduledPopup({
      orderDelivery: ownDriverDelivery,
    });

    const shiftInfo = shiftDetails(ownDriverDelivery, timeZone);
    const headerTxt = "Delivery scheduled for";
    const contentTxt = `${shiftInfo?.startTime} - ${shiftInfo?.endTime}`;

    // width < 300
    act(() => {
      global.innerWidth = 250;
      global.dispatchEvent(new Event("resize"));
    });
    await waitFor(() => expect(getByText(headerTxt)).toHaveStyle({fontSize: "16px"}));
    await waitFor(() => expect(getByText(contentTxt)).toHaveStyle({fontSize: "16px"}));

    // 300 < width <= 340
    act(() => {
      global.innerWidth = 340;
      global.dispatchEvent(new Event("resize"));
    });
    await waitFor(() => expect(getByText(headerTxt)).toHaveStyle({fontSize: "20px"}));
    await waitFor(() => expect(getByText(contentTxt)).toHaveStyle({fontSize: "20px"}));

    // width > 340
    act(() => {
      global.innerWidth = 341;
      global.dispatchEvent(new Event("resize"));
    });
    await waitFor(() => expect(getByText(headerTxt)).toHaveStyle({fontSize: "24px"}));
    await waitFor(() => expect(getByText(contentTxt)).toHaveStyle({fontSize: "24px"}));
  });

  it("should set fontSize of date and day text according to window inner width", async () => {
    const {getByText} = renderDeliveryScheduledPopup({
      orderDelivery: ownDriverDelivery,
    });

    const shiftInfo = shiftDetails(ownDriverDelivery, timeZone);
    const contentTxt = `${shiftInfo?.day}, ${shiftInfo?.month} ${shiftInfo?.date}`;

    // width < 300
    act(() => {
      global.innerWidth = 250;
      global.dispatchEvent(new Event("resize"));
    });
    await waitFor(() => expect(getByText(contentTxt)).toHaveStyle({fontSize: "12px"}));

    // 300 < width <= 320
    act(() => {
      global.innerWidth = 320;
      global.dispatchEvent(new Event("resize"));
    });
    await waitFor(() => expect(getByText(contentTxt)).toHaveStyle({fontSize: "16px"}));

    // width > 340
    act(() => {
      global.innerWidth = 341;
      global.dispatchEvent(new Event("resize"));
    });
    await waitFor(() => expect(getByText(contentTxt)).toHaveStyle({fontSize: "18px"}));
  });
});
