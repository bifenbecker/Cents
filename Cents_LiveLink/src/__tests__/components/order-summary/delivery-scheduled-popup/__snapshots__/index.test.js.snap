// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryScheduledPopup Component When deliveryProvider is ON DEMAND should match snaphot 1`] = `
<div>
  <div
    style="position: fixed; width: 0px; height: 0px; top: 0px; left: 0px; z-index: 99999999;"
  >
    <div
      style="position: fixed; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; opacity: 1; transition: opacity 0.2s ease-out; background: white;"
    />
    <div
      style="position: fixed; z-index: 1; box-shadow: 0 0 4px rgba(0, 0, 0, 0.3); background: white; left: 25%; top: 0px; width: 50%; height: 768px; transition: left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out;"
    >
      <div
        style="position: absolute; z-index: 2; opacity: 0; top: -5px; height: 10px; left: 0px; width: 100%; cursor: row-resize;"
      />
      <div
        style="width: 100%; height: 100%; overflow: auto;"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi hootTa hzPtvw"
        >
          <div
            class="sc-bczRLJ sc-gsnTZi guItVf hzPtvw"
          >
            <div
              class="sc-bczRLJ haqESh"
              color="CENTS_BLUE"
              font-size="18px"
              font-weight="700"
            >
              Your laundry is ready!
            </div>
            <div
              class="sc-bczRLJ sc-gsnTZi JBpjR hzPtvw"
            >
              <div
                class="sc-bczRLJ sc-gsnTZi iIJBxE hzPtvw"
              >
                <div
                  class="sc-bczRLJ eVqmpI"
                  color="BLACK"
                  font-size="24px"
                  font-weight="700"
                >
                  Delivery scheduled for
                </div>
                <div
                  class="sc-bczRLJ eVqmpI"
                  color="BLACK"
                  font-size="24px"
                  font-weight="700"
                >
                  08:00AM
                   - 
                  08:30AM
                </div>
                <div
                  class="sc-bczRLJ fhewhz"
                  color="BLACK"
                  font-family="secondary"
                  font-size="18px"
                >
                  Friday
                  , 
                  April
                   
                  22
                </div>
              </div>
              <div
                class="sc-bczRLJ kYYatg"
              >
                <img
                  alt="Car Image"
                  class="sc-bczRLJ iQjTFr"
                  src="Delivery_Car_Half.svg"
                />
              </div>
            </div>
            <div
              class="sc-bczRLJ dXmDhD"
              color="CENTS_BLUE"
              font-size="18px"
              font-weight="500"
            >
              See my delivery options
            </div>
          </div>
          <div
            class="sc-bczRLJ sc-gsnTZi kpIoXT hzPtvw"
          >
            <button
              class="sc-bczRLJ fgqTWn"
            >
              got it
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DeliveryScheduledPopup Component When deliveryProvider is OWN DRIVER should match snaphot 1`] = `
<div>
  <div
    style="position: fixed; width: 0px; height: 0px; top: 0px; left: 0px; z-index: 99999999;"
  >
    <div
      style="position: fixed; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; opacity: 1; transition: opacity 0.2s ease-out; background: white;"
    />
    <div
      style="position: fixed; z-index: 1; box-shadow: 0 0 4px rgba(0, 0, 0, 0.3); background: white; left: 25%; top: 0px; width: 50%; height: 768px; transition: left 0.2s ease-out,top 0.2s ease-out,width 0.2s ease-out,height 0.2s ease-out;"
    >
      <div
        style="position: absolute; z-index: 2; opacity: 0; top: -5px; height: 10px; left: 0px; width: 100%; cursor: row-resize;"
      />
      <div
        style="width: 100%; height: 100%; overflow: auto;"
      >
        <div
          class="sc-bczRLJ sc-gsnTZi hootTa hzPtvw"
        >
          <div
            class="sc-bczRLJ sc-gsnTZi guItVf hzPtvw"
          >
            <div
              class="sc-bczRLJ haqESh"
              color="CENTS_BLUE"
              font-size="18px"
              font-weight="700"
            >
              Your laundry is ready!
            </div>
            <div
              class="sc-bczRLJ sc-gsnTZi JBpjR hzPtvw"
            >
              <div
                class="sc-bczRLJ sc-gsnTZi iIJBxE hzPtvw"
              >
                <div
                  class="sc-bczRLJ eVqmpI"
                  color="BLACK"
                  font-size="24px"
                  font-weight="700"
                >
                  Delivery scheduled for
                </div>
                <div
                  class="sc-bczRLJ eVqmpI"
                  color="BLACK"
                  font-size="24px"
                  font-weight="700"
                >
                  09:00AM
                   - 
                  02:00PM
                </div>
                <div
                  class="sc-bczRLJ fhewhz"
                  color="BLACK"
                  font-family="secondary"
                  font-size="18px"
                >
                  Thursday
                  , 
                  April
                   
                  21
                </div>
              </div>
              <div
                class="sc-bczRLJ kYYatg"
              >
                <img
                  alt="Van Image"
                  class="sc-bczRLJ iQjTFr"
                  src="Blue_Van_Half.svg"
                />
              </div>
            </div>
            <div
              class="sc-bczRLJ dXmDhD"
              color="CENTS_BLUE"
              font-size="18px"
              font-weight="500"
            >
              See my delivery options
            </div>
          </div>
          <div
            class="sc-bczRLJ sc-gsnTZi kpIoXT hzPtvw"
          >
            <button
              class="sc-bczRLJ fgqTWn"
            >
              got it
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
